src/python/pants/backend/helm/util_rules/tool.py

```diff
@@ -183,14 +441,16 @@ def helm_process(request: HelmProcess, helm_binary: HelmBinary) -> Process:
         **request.extra_immutable_input_digests,
     }
 
+    append_only_caches = {**helm_binary.append_only_caches, **request.extra_append_only_caches}
+
     return Process(
         [helm_binary.path, *request.argv],
         input_digest=request.input_digest,
         immutable_input_digests=immutable_input_digests,
         env=env,
         description=request.description,
         level=request.level,
-        append_only_caches=helm_binary.append_only_caches,
+        append_only_caches=append_only_caches,
         output_directories=request.output_directories,
         output_files=request.output_files,
         cache_scope=request.cache_scope or ProcessCacheScope.SUCCESSFUL,
```

Comment:
Probably best to let this default to the underlying default rather than specifying it explicitly (not that we're planning on changing it anytime soon).

Reply:
I'm specifying it because of two things:
- the `helm_binary` needs an additional append only cache, otherwise the process that does `helm pull` when resolving 3rd party artifacts fails
- the the post-processor needed during the deployment process (created from a `VenvPexProcess`) has its own append only caches

Is there a way to provide this to the `Process` type without overriding the defaults?

Reply:
Ah, sorry: I was trying to comment on the `cache_scope` line. Could do:
```suggestion
        cache_scope=request.cache_scope,
```

Reply:
oh I see, can't simply assign it as `cache_scope` in `Process` has type `ProcessCacheScope`, while that field in `HelmProcess` is `Optional[ProcessCacheScope]`.

Could change the type of the field in `Process` so it becomes optional but maybe better doing it in separate PR?